### PR TITLE
separate sfrac config options

### PR DESCRIPTION
**Pull request by eclee25**

### CODE REVIEW

File: Analysis/R/prepare_covar_cube.R [hunk 1]
```diff
@@ -26,27 +30,28 @@ prepare_covar_cube <- function(
     res_time,
     username,
     covariate_transformations,
-    sfrac_thresh
+    sfrac_thresh,
+    sfrac_thresh_lp
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:34]: From what I get sfrac_thresh is used for the output shapefiles, and sfrac_thresh_lp is used for in data.

I thought that the intention was to use one value for borders, and one for connections. If this is correct, I would suggest re-naming to sfrac_thresh_border and sfrac_thresh_conn.

The next comments indicate when one and the other should  be used.

File: Analysis/R/prepare_covar_cube.R [hunk 2]
```diff
@@ -120,126 +125,126 @@ prepare_covar_cube <- function(
...
   # Get pixels with low sfrac
-  low_sfrac <- location_periods_dict %>% 
-    dplyr::group_by(rid, x, y) %>% 
-    dplyr::slice_max(pop_weight) %>% 
-    dplyr::filter(pop_weight < sfrac_thresh) %>% 
-    dplyr::select(rid, x, y) %>% 
+  low_sfrac <- location_periods_dict %>%
+    dplyr::group_by(rid, x, y) %>%
+    dplyr::slice_max(pop_weight) %>%
+    dplyr::filter(pop_weight < sfrac_thresh_lp) %>%
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:191]: sfrac_thresh_border

File: Analysis/R/prepare_covar_cube.R [hunk 3]
```diff
@@ -120,126 +125,126 @@ prepare_covar_cube <- function(
...
+    dplyr::select(rid, x, y) %>%
     dplyr::inner_join(sf_grid %>% sf::st_drop_geometry())
-  
+
   if (nrow(low_sfrac) > 0) {
     cat("---- Dropping", nrow(low_sfrac), "space grid cells because the max sfrac is below",
-        sfrac_thresh, ". \n")
-    
+        sfrac_thresh_lp, ". \n")
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:197]: sfrac_thresh_border

File: Analysis/R/prepare_covar_cube.R [hunk 4]
```diff
@@ -120,126 +125,126 @@ prepare_covar_cube <- function(
...
+
     # Re-define grid cells to remove cells with low sf_frac
     non_na_gridcells <- setdiff(non_na_gridcells, low_sfrac$long_id)
-    
+
   } else {
-    cat("---- No cells with sfrac below", sfrac_thresh, ".\n")
+    cat("---- No cells with sfrac below", sfrac_thresh_lp, ".\n")
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:203]: sfrac_thresh_border

File: Analysis/R/prepare_covar_cube.R [hunk 5]
```diff
@@ -120,126 +125,126 @@ prepare_covar_cube <- function(
...
   # Drop grid cells to location periods connections
   location_periods_dict <- location_periods_dict %>%
     dplyr::mutate(connect_id = dplyr::row_number())
-  
-  low_sfrac_connections <- location_periods_dict %>% 
-    dplyr::filter(pop_weight < sfrac_thresh)
-  
+
+  low_sfrac_connections <- location_periods_dict %>%
+    dplyr::filter(pop_weight < sfrac_thresh_lp)
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:227]: sfrac_thresh_conn

File: Analysis/R/prepare_covar_cube.R [hunk 6]
```diff
@@ -120,126 +125,126 @@ prepare_covar_cube <- function(
...
+
   cat("Dropping", nrow(low_sfrac_connections), "/", nrow(location_periods_dict),
       "connections between grid cells",
-      "and location periods which have sfrac <", sfrac_thresh,  "\n")
-  
-  location_periods_dict <- location_periods_dict %>% 
+      "and location periods which have sfrac <", sfrac_thresh_lp,  "\n")
```
Comment (javierps) [inline Analysis/R/prepare_covar_cube.R:231]: sfrac_thresh_conn

**Review by javierps — CHANGES_REQUESTED** [review]
The separation would also need to be done for the new dropping of output shapefile connections here: https://github.com/HopkinsIDD/cholera-mapping-pipeline/blob/c5fdc11fb67e048ef4d5c57442dc1206f9f6eb61/Analysis/R/prepare_stan_input.R#L489

File: Analysis/R/prepare_stan_input.R [hunk 7]
```diff
@@ -484,158 +484,158 @@ prepare_stan_input <- function(
...
   # Get pixels with low sfrac
-  output_low_sfrac <- output_location_periods_table %>% 
-    dplyr::group_by(rid, x, y) %>% 
-    dplyr::slice_max(pop_weight) %>% 
-    dplyr::filter(pop_weight < sfrac_thresh) %>% 
-    dplyr::select(rid, x, y) %>% 
+  output_low_sfrac <- output_location_periods_table %>%
+    dplyr::group_by(rid, x, y) %>%
+    dplyr::slice_max(pop_weight) %>%
+    dplyr::filter(pop_weight < sfrac_thresh_conn) %>%
```
Comment (javierps) [inline Analysis/R/prepare_stan_input.R:492]: This should be sfrac_thresh_border.

**Review by javierps — APPROVED** [review]

File: Analysis/R/prepare_stan_input.R (hunk 7 above)
Comment (eclee25) [reply Analysis/R/prepare_stan_input.R:516]: I’m not sure I understand why that line should be changed to sfrac_thresh_border. Isn’t the pixel linkage to an output shapefile more equivalent to the sfrac_thresh_conn setting? This isn’t going to affect the appearance of any grid cell estimate in our outputs

Comment (javierps) [reply Analysis/R/prepare_stan_input.R:516]: I see what you mean. Also, thinking about it, if the national shapefile for observations is the same as the one for the output, then this dropping part should not be necessary, and the connection part after should take care of everything. So we can either keep here sfrac_thresh_conn as you suggest, or remove all lines 513-523.

**Pull request merged**